---
title: "Class 09: Structural Bioinformatics"
author: "Eric Jordahl"
format: html
date: 2022-10-26
toc: True
---


## Visualizing the HIV-1 protease structure

# The importance of water

# **Q4**

**Water molecules normally have 3 atoms. Why do we see just one atom per water molecule in this structure?**

We can only see the oxygen atom because the resolution on this viewer is too large for the hydrogen to be shown.

# **Q5**

**There is a critical "conserved" water molecule in the binding site. Can you identify this water molecule? What residue number does this water molecule have?**

The hydrogen bond of the water is able to interact with both Asp25 residues which can bind better than the ligand (when it does not have the ability to H-bond)

# **Q6**

**Generate and save a figure clearly showing the two distinct chains of HIV-protease along with the ligand. You might also consider showing the catalytic residues ASP 25 in each chain and the critical water (we recommend "Ball & Stick" for these side-chains). Add this figure to your Quarto document.**

![](1HSG%20(2))

# Reading and working with strutures in R

The `bio3d` package for structural bioinformatics has lots of features for reading adn working with bimolecular sequences and structures.


```{r}
library (bio3d)
pdb <- read.pdb("1HSG")
pdb
```


# **Q7**

**How many amino acid residues are there in this pdb object?**

There are 198 amino acids in the protein,

# **Q8**

**Name one of the two non-protein residues?**

MK1 - Merck 1

# **Q9**

**How many protein chains are in this structure?**

There are 2 chains in this structure


```{r}
head(pdb$atom)
```

```{r}
adk <- read.pdb("6s36")
adk
```


Normal mode analysis (NMA) is a bioinformatics method for predicting functional motions, It will show us the parts of the protein that are "flexible"


```{r}
m <- nma(adk)
plot(m)
```


Make a "movie" of this thing moving.


```{r}
mktrj(m, file="adk_m7.pdb")
```


# Comparative Analysis of all ADK structures

First we get the sequence of ADK and use this to search the PDB database.

Software Installs


```{r}
#install.packages("bio3d")
#install.packages("devtools")
#install.packages("BiocManager")

#BiocManager::install("msa")
#devtools::install_bitbucket("Grantlab/bio3d-view")
```


# **Q10**

**Which of the packages above is found only on BioConductor and not CRAN?**

BiocManager

# **Q11**

**Which of the above packages is not found on BioConductor or CRAN?**

BitBucket

# **Q12**

**True or False? Functions from the devtools package can be used to install packages from GitHub and BitBucket?**

True


```{r}
library(bio3d)
aa <- get.seq("1ake_A")
aa
```


# **Q13**

**How many amino acids are in this sequence, i.e. how long is this sequence?**

There are 214 amino acids in this sequence


```{r}
b <- blast.pdb(aa)
```

```{r}
hits <- plot(b)
```

```{r}
hits$pdb.id
```

```{r}
pdb.annotate(hits$pdb.id)
```

```{r}
files <- get.pdb(hits$pdb.id, path="pdbs", split=TRUE, gzip=TRUE)
```


Viewing all these structures looks like a hot mess! We need to try something else...

We will align and supperpose these structures.


```{r}
pdbs <- pdbaln(files, fit = TRUE, exefile="msa")

# Vector containing PDB codes for figure axis
ids <- basename.pdb(pdbs$id)

# Draw schematic alignment
plot(pdbs, labels=ids)
```


# Annotate PDB Structures


```{r}
anno <- pdb.annotate(ids)
unique(anno$source)
anno
```


# Principal Component Analysis


```{r}
# Perform PCA
pc.xray <- pca(pdbs)
plot(pc.xray)
```

```{r}
# Calculate RMSD
rd <- rmsd(pdbs)

# Structure-based clustering
hc.rd <- hclust(dist(rd))
grps.rd <- cutree(hc.rd, k=3)

plot(pc.xray, 1:2, col="grey50", bg=grps.rd, pch=21, cex=1)
```

```{r}
pc1 <- mktrj(pc.xray, pc=1, file="pc_1.pdb")
```


![](PC_1.PDB_animate-trajectory.mp4)


```{r}
#Plotting results with ggplot2
library(ggplot2)
library(ggrepel)

df <- data.frame(PC1=pc.xray$z[,1], 
                 PC2=pc.xray$z[,2], 
                 col=as.factor(grps.rd),
                 ids=ids)

p <- ggplot(df) + 
  aes(PC1, PC2, col=col, label=ids) +
  geom_point(size=2) +
  geom_text_repel(max.overlaps = 20) +
  theme(legend.position = "none")
p
```


# Normal Mode Analysis


```{r}
# NMA of all structures
modes <- nma(pdbs)
plot(modes, pdbs, col=grps.rd)
```


# **Q14**

**What do you note about this plot? Are the black and colored lines similar or different? Where do you think they differ most and why?**

The black lines and colored lines share similar shapes, but the colored lines fluctuate more than the black lines and the differ most around the 130-150 range where the colored lines really spike more than the black ones. These residues correspond well to the ones we saw "moving" in the animated version on Mol Viewer.

